services/mgmt/compute/examples/vm_list_using_resource_graph.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::Arc;
+use azure_identity::DefaultAzureCredential;
+use azure_mgmt_resourcegraph::models::{QueryRequest, QueryRequestOptions};
+use serde_json::{json, Value};
+use azure_mgmt_resourcegraph::Client as AzureResourceGraphClient;
+use azure_mgmt_resourcegraph::ClientBuilder as AzureResourceGraphClientBuilder;
+
+/// Fetching Virtual Machines using Azure Resource Graph
+
+#[derive(Debug)]
+pub enum GenericError {
+    InvalidRequest,
+}
+
+#[derive(Debug)]
+struct CustomError {
+    err_type: GenericError,
+    err_msg: String
+}
+
+async fn fetch_azure_records() -> Result<Value, CustomError> {
+
+    let azure_creds = Arc::new(DefaultAzureCredential::default());
+
+    let az_rg_client = AzureResourceGraphClient::builder(azure_creds).build();
+
+    // fetch first 10 records
+    let query_options = QueryRequestOptions {
+        top: Option::from(10),
+        skip: Option::from(0),
+        ..Default::default()
+    };
+
+    // in the below request, we are fetching resources of type (virtual machines)
+    let custom_query_request = QueryRequest {
+        subscriptions: vec![],
+        management_groups: vec![],
+        query: "Resources | where type == 'microsoft.compute/virtualmachines'".to_string(),
+        options: Option::from(query_options),
+        facets: vec![],
+    };
+
+    let query_request = QueryRequest::from(custom_query_request);
+
+    let resources = az_rg_client.resources(query_request).await;
+
+    let result = match resources {
+        Ok(d) => {
+            let elements = d.data.as_array().unwrap();
+            let obj = json!(elements);
+            obj
+        }
+        Err(e) => {
+            let actual_error_message = format!("could not fetch data from azure resource graph : {:?}", e.to_string());
+            let custom_err = CustomError {
+                err_msg: actual_error_message,
+                err_type: GenericError::InvalidRequest
+            };
+            return Err(custom_err)
+        }
+    };
```

Comment:
Instead of reserializing these to json for an example, perhaps this could be simplified to show a single entry for each record that is returned?  Something akin to this?

```
    let response = client.resources(query_request).await?;
    println!("total records: {}", response.total_records);
    if let Some(as_array) = response.data.as_array() {
        for entry in as_array {
            let resource_id = entry.get("id");
            println!("{resource_id:?}");
        }
    }
```

This would print something akin to:
```
total records: N
Some(String("/subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Compute/VirtualMachines/VM_NAME"))
Some(String("/subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Compute/VirtualMachines/VM_NAME"))
Some(String("/subscriptions/SUBSCRIPTION_ID/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Compute/VirtualMachines/VM_NAME"))
```

Reply:
This also works.